Bling and StickRumble fixes and additions

KyleRAnderson opened the pull request:
Optimized the bling susbsytem for battery displaying, added a battery level calculator (very approximate). StickRumble command also has an added interval feature and works as it should now. 

*All has been tested on plyboy. Bling subsystem will not work on Mergius, therefore meaning the battery indications must be turned off.*

KyleRAnderson committed:
Addition of functional Stick Rumble patterns

*Tested* The Stick Rumbling Pattern command now functions properly, and
some minor changes were made to the bling system.

Everything here is tested.

KyleRAnderson committed:
Merge remote-tracking branch 'refs/remotes/FRC2706/master'

# Conflicts:
#	src/org/usfirst/frc/team2706/robot/Robot.java

KyleRAnderson committed:
Addition of BatteryReader

The battery reader, which will read battery level has been tested and
now added.

Small tweaks were made to the bling subsystem and the intervals feature
was added to the StickRumble commmand.

KyleRAnderson committed:
Merged with additions in #37

KyleRAnderson committed:
Quick Look Changes

ryanlarkin commented on src/org/usfirst/frc/team2706/robot/RobotMap.java:
```diff
@@ -162,6 +162,13 @@
 
+    // Bling Various
+    // Number of pixels in LED Strip
+    public static final int LED_Strip_PixelCount = 120;
```
This isn't really how RobotMap is meant to be used.
The RobotMap constants end before `ROBOT_ID_LOC`
```Java
    // Bling Various
    // Number of pixels in LED Strip
    private static final int[] LED_STRIP_PIXELCOUNT_VALS = {120, 120, 120};
    public static final int LED_STRIP_PIXELCOUNT = getConstant("LED_STRIP_PIXELCOUNT ");
   
    // Battery Variable which equals the voltage of the battery
    private static final double[] BATTERY_CAPACITY_VALS = {12.0, 12.0, 12.0};
    public static final double BATTERY_CAPACITY = getConstant("BATTERY_CAPACITY ");
```

ryanlarkin commented on src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java:
```diff
@@ -0,0 +1,77 @@
+
+    public BatteryReader() {
+
+        pdp = new PowerDistributionPanel();
```
Try spacing these out to make them more readable, or add comments

ryanlarkin commented on src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java:
```diff
@@ -0,0 +1,77 @@
+
+    @Override
+    public void end() {
+
```
Try to remove this if you can, if not just put the brackets like `{}`

ryanlarkin commented on src/org/usfirst/frc/team2706/robot/controls/StickRumble.java:
```diff
@@ -0,0 +1,163 @@
+
+    public static int repeatCountCopy;
+
+    public static boolean intervalOn = false; // Used to determine if we are in the interval time
```
Try to put comments above code to make it more readable

ryanlarkin commented on src/org/usfirst/frc/team2706/robot/controls/StickRumble.java:
```diff
@@ -0,0 +1,163 @@
+    /**
+     * Simply to make it easier to rumble certain sticks.
+     * 
+     * @param on : Determines whether we're turning the rumble on or off.
```
You don't need a : there

ryanlarkin commented on src/org/usfirst/frc/team2706/robot/subsystems/Bling.java:
```diff
@@ -4,20 +4,23 @@
     SerialPort blingPort = new SerialPort(9600, SerialPort.Port.kMXP);
 
-    int pixels = 120; // The number of pixels on one LED strip
+    int pixels = RobotMap.LED_Strip_PixelCount; // The number of pixels on one LED strip
```
Try to put the comment above the line

ryanlarkin commented on src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java:
```diff
@@ -0,0 +1,77 @@
+
+    public static double fullBatteryCapacity = RobotMap.BatteryCapacity;
+
+    public static PowerDistributionPanel pdp; // Get a new battery object.
```
It would probably be easier to read if the comment was on the line above.

ryanlarkin commented on src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java:
```diff
@@ -0,0 +1,77 @@
+         * don't want to spam the bling system.
+         */
+        if (batteryPercent <= 0.2 && !batCritical) {
+            System.out.println("Low Battery " + batCritical);
```
Will this spam the RIOLog? That should probably be avoided if it does. Maybe print it every 5 minutes.

KyleRAnderson replied:
No, it wouldn't because batCritical turns to true the first time this occurs so it wouldn't run a second time. Do you actually want this or do you want a more prominent warning? I could even set a rumble on the controllers.

ryanlarkin replied:
Maybe something like `DriverStation.getInstance().reportWarning("Low Battery " + batCritical", false);`. That would print directly to the driver station

ryanlarkin commented on src/org/usfirst/frc/team2706/robot/controls/StickRumble.java:
```diff
@@ -0,0 +1,163 @@
+     */
+    public boolean isFinished() {
+
+        if (intervalCount <= 0) { // If on is set to off anywhere (false) then we
```
Try to put the comment above.

ryanlarkin commented on src/org/usfirst/frc/team2706/robot/controls/StickRumble.java:
```diff
@@ -0,0 +1,163 @@
+    @Override
+    public void end() {
+
+        System.out.println("Ended the stick rumble forcefully");
```
This is called even when `isFinished()` returns true, so it isn't really ended forcefully

ryanlarkin reviewed (dismissed):
There are a couple of small issues with this pull request.
~1. How often are you printing to the RIO Log, we should only print sometimes, and if it is something you want to get the attention of the driver with, you should use DriverStation.warning() or something~
~2. Try to use RobotMap like the rest of the constants, and follow proper formatting for constants `LIKE_THIS`.~
3. There are some misc. formatting errors. (I can fix this up) (Made PR in your fork)
4. There are a lot of unnecessary empty lines. (I can fix this up) (Made PR in your fork)

KyleRAnderson committed:
Changes for #38 pt. 1

Fixed up everything outlined in #388

cnnradams commented on src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java:
```diff
@@ -0,0 +1,76 @@
+
+
+    public static double batteryOutputVoltage;
+    // Used to tell us if the battery level is critical
```
There should be a space right before the comment line so we know which line its referring to and it looks nicer

KyleRAnderson replied:
Completed this in 8fcbc81aa2feffc00c20925578919f871f3e0090.

KyleRAnderson committed:
Fixes for #38 pt. 2

Just one small fix and the added driverstaton warning.

KyleRAnderson committed:
Added exception handling

If there is hardware missing, the bling subsystem will handle it by
returning in all of its commands before it causes any erros.

ryanlarkin commented on src/org/usfirst/frc/team2706/robot/RobotMap.java:
```diff
@@ -162,6 +162,10 @@
 
+    
+    // Battery Variable which equals the voltage of the battery
+    public static final double BatteryCapacity = 12.0;
```
To use RobotMap properly, this should go with the rest of the constants before `ROBOT_ID_LOC` and look like:
```Java
    // Battery Variable which equals the voltage of the battery
    private static final double[] BATTERY_CAPACITY_VALS = {12.0, 12.0, 12.0};
    public static final double BATTERY_CAPACITY = getConstant("BATTERY_CAPACITY");
```

KyleRAnderson replied:
Just got rid of that since it wasn't necessary for it to be there.

ryanlarkin reviewed (dismissed):
Just need to fix RobotMap, and I can fix any minor formatting issues.

KyleRAnderson committed:
Fixes for #38 pt. 3

ryanlarkin committed:
Fix minor formatting issues

ryanlarkin committed:
Other minor fixes

ryanlarkin committed:
Merge branch 'master' into master

ryanlarkin committed:
Merge branch 'master' into formatting

KyleRAnderson committed:
Bling and StickRumble additions

StickRumble is now directly part of the main robot code, and has an
initialization pattern for autonomous and teleop modes.

Bling system has additions for teleop performance.
*Changes not tested*

KyleRAnderson committed:
Merge remote-tracking branch 'refs/remotes/FRC2706/master'

Conflicts werre resolved.

KyleRAnderson committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/org/usfirst/frc/team2706/robot/Robot.java
#	src/org/usfirst/frc/team2706/robot/subsystems/Bling.java

KyleRAnderson committed:
Merge remote-tracking branch 'refs/remotes/FRC2706/master'

# Conflicts:
#	src/org/usfirst/frc/team2706/robot/subsystems/GearHandler.java

KyleRAnderson commented:
Need to wait on #48 to have the new features outlined.

ryanlarkin commented on .classpath:
```diff
@@ -11,7 +11,5 @@
 			<attribute name="javadoc_location" value="jar:platform:/resource/2017-2706-Robot-Code/lib/non-robot-libs/google/docs.zip!/"/>
 		</attributes>
 	</classpathentry>
-	<classpathentry kind="var" path="USERLIBS_DIR/CTRLib.jar"/>
```
These shouldn't be removed, otherwise the talons and NavX will stop working

KyleRAnderson replied:
Eclipse did it automatically. I don't seem to have the proper modules for them installed. How can I get them installed?

KyleRAnderson replied:
Nevermind. Figured it out.

ryanlarkin committed:
Merge branch 'master' into formatting

ryanlarkin reviewed (dismissed):
The NavX and CAN Talon libraries need to be re-added to the classpath. Also I have a pull request open to fix some formatting stuff in your fork.

KyleRAnderson committed:
Fixing accidents with the .classpath

ryanlarkin commented:
Just deploy the code, you don't need to be connected to the robot, then
restart eclipse

On Feb 26, 2017 1:54 PM, "Kyle Anderson" <notifications@github.com> wrote:

> *@eAUE* commented on this pull request.
> ------------------------------
>
> In .classpath
> <https://github.com/FRC2706/2017-2706-Robot-Code/pull/38#discussion_r103112814>
> :
>
> > @@ -11,7 +11,5 @@
>  			<attribute name="javadoc_location" value="jar:platform:/resource/2017-2706-Robot-Code/lib/non-robot-libs/google/docs.zip!/"/>
>  		</attributes>
>  	</classpathentry>
> -	<classpathentry kind="var" path="USERLIBS_DIR/CTRLib.jar"/>
>
> Eclipse did it automatically. I don't seem to have the proper modules for
> them installed. How can I get them installed?
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/FRC2706/2017-2706-Robot-Code/pull/38#discussion_r103112814>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFhS7ZsLjGOaxI-jS0vi_rqLY3O2j__-ks5rgcpqgaJpZM4MFPke>
> .
>

cnnradams commented on src/org/usfirst/frc/team2706/robot/subsystems/GearHandler.java:
```diff
@@ -100,5 +105,7 @@ public boolean checkArmOpen() {
         return false;
     }
+    
+
```
No spaces here

KyleRAnderson replied:
Done

cnnradams reviewed (approved):
I'm happy with it maybe wait for Ryan if he wants to look a little deeper? Nice documentation by the way 👍

ryanlarkin commented on .classpath:
```diff
@@ -11,7 +11,7 @@
-	<classpathentry kind="var" path="USERLIBS_DIR/CTRLib.jar"/>
-	<classpathentry kind="var" path="USERLIBS_DIR/navx_frc.jar"/>
+	<classpathentry kind="lib" path="build/jars/navx_frc.jar"/>
+	<classpathentry kind="lib" path="build/jars/CTRLib.jar"/>
```
This will not deploy correctly, it needs to be exactly the same as it was before.

KyleRAnderson replied:
Sorry. Fixed it in 35e910c

ryanlarkin reviewed (changes requested):
The .classpath still isn't correct, also could you take a look at https://github.com/eAUE/2017-2706-Robot/pull/1

KyleRAnderson committed:
More Quick fixes for #38

KyleRAnderson committed:
Merge pull request #1 from eandr127/formatting

Fix minor formatting issues

ryanlarkin reviewed (changes requested):
Instead of adding a teleopInit, auto, teleopPeriodic, clear to Bling, make a command that can do this. You can make it the default command. You can use DriverStation.isTelop() if you want too.

KyleRAnderson commented:
I made a seperate command to replace teleopPeriodic, but clear, teleopInit and auto are all only run once so I didn't think it was worth it.

KyleRAnderson committed:
Changes for #38 pt.

Just quick changes. Added a BlingTeleop command to replace the
Bling.teleopPeriodic command.

KyleRAnderson committed:
Merge remote-tracking branch 'refs/remotes/FRC2706/master'

ryanlarkin committed:
Fix formatting, move command to subsystem

ryanlarkin commented:
If possible, I'd still want to keep Robot as minimal as possible, so I made a small change to the command, not sure if it works though. eAUE#2

KyleRAnderson committed:
Feature additions to Bling and StickRumble

- Added an infinite rumble feature in the StickRumble program.
- Added better indications for the gear system.
- BlingTeleop now uses the new features from the GearHandler subsystem.

KyleRAnderson committed:
Merge pull request #2 from eandr127/subsystem-command

Fix formatting, move command to subsystem *Untested*

KyleRAnderson committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/org/usfirst/frc/team2706/robot/commands/teleop/BlingTeleop.java

ryanlarkin commented on src/org/usfirst/frc/team2706/robot/Robot.java:
```diff
@@ -107,7 +111,9 @@ public void robotInit() {
      */
-    public void disabledInit() {}
+    public void disabledInit() {
+        blingSystem.clear(); // S.V.P leave this here. 
```
In the Bling command, when it is canceled or ended, it should cancel automatically.

KyleRAnderson replied:
That command doesn't cancel, it tells the arduino to clear the LED strip. Otherwise, the pattern will stay on. I guess we could leave it to the command to do that in the `end()` and `interrupted()` commands. I'll change that. Done in f0f2e88.

KyleRAnderson committed:
Changes for #38 pt. 5

ryanlarkin reviewed (approved):
Looks good

ryanlarkin committed:
Merge branch 'master' into master

ryanlarkin committed:
Merge branch 'master' into master

ryanlarkin committed:
Merge branch 'master' into master

ryanlarkin commented:
@eAUE, is this ready to be merged, or does it need testing?

KyleRAnderson commented:
It needs to be tested. I'm going to be at the KRP to do that today.

KyleRAnderson committed:
Quick Commit to move on

*Untested*

KyleRAnderson committed:
Merge with remote tracking branch

KyleRAnderson committed:
Merge remote-tracking branch 'refs/remotes/FRC2706/master'

KyleRAnderson commented:
@eandr127 @Stealth2994 can you guys put the new robot code with the new controller mappings on GitHub by the way for drive practise later today? Thanks!

ryanlarkin commented:
It's currently in GearWallAlignment branch, I can hopefully have it merged
into master by 1.

On Mar 5, 2017 11:41 AM, "Kyle Anderson" <notifications@github.com> wrote:

> @eandr127 <https://github.com/eandr127> @Stealth2994
> <https://github.com/Stealth2994> can you guys put the new robot code with
> the new robot mappings on GitHub by the way for drive practise later today?
> Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/FRC2706/2017-2706-Robot-Code/pull/38#issuecomment-284242002>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFhS7e6z2mSGvziWSuwQiBMGl5HSuVMZks5riuWxgaJpZM4MFPke>
> .
>

KyleRAnderson committed:
Quick fixes

KyleRAnderson committed:
Merge remote-tracking branch 'refs/remotes/FRC2706/GearWallAlignment'

KyleRAnderson committed:
Merge remote-tracking branch 'refs/remotes/FRC2706/practice-code'

# Conflicts:
#	src/org/usfirst/frc/team2706/robot/OI.java

KyleRAnderson committed:
Resolved #59

The weird patterns no longer appear - tested.
Haven't tested the battery indication yet, however this will be easy
enough to test.

KyleRAnderson committed:
More resolution for #59

KyleRAnderson commented:
@eandr127 @Stealth2994 so all of my code is working now, but I have the new stuff from the practice branch in here as well. I'll wait for you guys to merge that before I merge this.

ryanlarkin committed:
Merge branch 'master' into master

KyleRAnderson commented on src/org/usfirst/frc/team2706/robot/Robot.java:
```diff
@@ -130,7 +130,13 @@ public void disabledPeriodic() {
+        camera.enableRingLight(true);
+
+        // Get the bling doing autonomous patterns.
+        blingSystem.auto(); 
```
I'm going to get the `BlingPeriodic` command doing this, no need to get the `autonomousInit` function doing it.

KyleRAnderson committed:
Merge remote-tracking branch 'refs/remotes/FRC2706/master'

# Conflicts:
#	src/org/usfirst/frc/team2706/robot/Robot.java
#	src/org/usfirst/frc/team2706/robot/subsystems/GearHandler.java

KyleRAnderson committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/org/usfirst/frc/team2706/robot/Robot.java
#	src/org/usfirst/frc/team2706/robot/subsystems/GearHandler.java

KyleRAnderson committed:
Stuff that didn't get merged

Just didn't merge some of the last commits' mess.

KyleRAnderson committed:
Formatting fixes and efficiency

Made small tweaks to formatting and made the code more efficient.

cnnradams commented:
I think this is ready to be merged... lol :)

KyleRAnderson commented:
Yeah, just before we do that how accurate is the IR sensor at detecting when the peg is in the hole?

KyleRAnderson commented:
I guess it doesn't really matter. I'll merge it.
